### ISSUE TITLE
Setup problem? ../Encoder is missing

### ISSUE THREAD

**Issue by Tux** [Sereal/Sereal#119]
```
.cpan/build/Sereal-3.009-lQWLsu$ perl Makefile.PL
failed to use blib: ../Encoder is not a directory
BEGIN failed--compilation aborted at (eval 9) line 1.
```

**Comment by demerphq** [comment]
On 1 December 2015 at 09:03, H.Merijn Brand <notifications@github.com>
wrote:

> .cpan/build/Sereal-3.009-lQWLsu$ perl Makefile.PL
> failed to use blib: ../Encoder is not a directory
> BEGIN failed--compilation aborted at (eval 9) line 1.
>
>
That code should only fire if you are building in a tree that looks like
our Repo.

IOW, this code is firing:

my $shared_dir = "../shared";
my $in_source_repo = -d "../../.git" and -d $shared_dir;

my $module = "Sereal";

unshift @INC, ".", "./inc";
unshift @INC, $shared_dir, "$shared_dir/inc" if $in_source_repo;

if ($in_source_repo) {
    eval "use blib '../Encoder'; use blib '../Decoder'; 1"
        or die "failed to use blib: $@";
}

I dont think we ever contemplated that someone would have a git repo two
levels up, and a shared directory one level up and not be in the Sereal
repo.

I will put together a more robust way to detect we are in the Sereal repo,
but for now, I think you should be able to do your build from a different
directory and avoid this problem.

Alternatively if you *are* building from the Sereal repo then you should do
a make cycle in the Decoder/Encoder dirs before you build Sereal.

Can you tell me which of these cases you have hit?

Yves

**Comment by Tux** [comment]
I have a .git in my ~/.cpan to maintain my preferences and patches: https://github.com/Tux/cpanprefs
```
$ cpan Sereal
```
will build in ~/.cpan/build/Sereal-x.yyy-zzzzzz/ so ../../.git will always be there

**Comment by demerphq** [comment]
What about ../shared ?

You also have that in your .cpan dir?

Yves

On 1 December 2015 at 10:47, H.Merijn Brand <notifications@github.com>
wrote:

> I have a .git in my ~/.cpan to maintain my preferences and patches:
> https://github.com/Tux/cpanprefs
>
> $ cpan Sereal
>
> will build in ~/.cpan/build/Sereal-x.yyy-zzzzzz/ so ../../.git will always
> be there
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Sereal/Sereal/issues/119#issuecomment-160915036>.
>



-- 
perl -Mre=debug -e "/just|another|perl|hacker/"

**Comment by Tux** [comment]
nope

**Comment by demerphq** [comment]
On 1 December 2015 at 11:03, H.Merijn Brand <notifications@github.com>
wrote:

> nope
>

Yeah, I grok it now. Precedence error in the following code:

my $shared_dir = "../shared";
my $in_source_repo = -d "../../.git" and -d $shared_dir;

It doesnt require that $shared_dir is there. :-(

I will  push a fix momentarily.

Yves

**Comment by demerphq** [comment]
This should be fixed in the latest version (along with #118). Thanks for  the report. Closing the ticket.
